codalab/worker/local_run/local_run_state.py

```diff
@@ -407,12 +408,18 @@ def _transition_from_CLEANING_UP(self, run_state):
         """
         bundle_uuid = run_state.bundle['uuid']
         if run_state.container_id is not None:
-            while True:
+            while docker_utils.container_exists(run_state.container):
                 try:
                     finished, _, _ = docker_utils.check_finished(run_state.container)
                     if finished:
                         run_state.container.remove(force=True)
+                        run_state = run_state._replace(container=None, container_id=None)
                         break
+                    else:
+                        try:
+                            run_state.container.kill()
+                        except docker.errors.APIError:
+                            logger.error(traceback.format_exc())
```

Comment:
Why does this not have a sleep like the following `except`?

Reply:
Good point, added.